packages/experimental/Menu/src/MenuItemRadio/useMenuItemRadio.ts

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+import { AccessibilityActionEvent, AccessibilityState } from 'react-native';
+import { MenuItemRadioProps, MenuItemRadioState } from './MenuItemRadio.types';
+import { memoize } from '@fluentui-react-native/framework';
+import {
+  InteractionEvent,
+  useAsPressable,
+  useKeyProps,
+  useOnPressWithFocus,
+  useViewCommandFocus,
+} from '@fluentui-react-native/interactive-hooks';
+import { useMenuListContext } from '../context/menuListContext';
+
+const defaultAccessibilityActions = [{ name: 'Toggle' }];
+
+export const useMenuItemRadio = (props: MenuItemRadioProps): MenuItemRadioState => {
+  // attach the pressable state handlers
+  const defaultComponentRef = React.useRef(null);
+  const {
+    accessibilityActions,
+    accessibilityState,
+    componentRef = defaultComponentRef,
+    disabled,
+    name,
+    onAccessibilityAction,
+    ...rest
+  } = props;
+  const context = useMenuListContext();
+  const checked = context.checked?.[name];
+  const selectRadio = context.selectRadio;
+
+  const toggleChecked = React.useCallback(
+    (e: InteractionEvent) => {
+      selectRadio(e, name, !checked);
```

Comment:
Here's the main difference from MenuItemCheckbox

Reply:
Would it be worth sharing the same hook and somehow passing a boolean`isRadio`? Either through the Context or the prop list? I'm ok if it makes sense to duplicate code, jus thought I'd ask.

Reply:
Sure, we can do that. It'd be something that I'd have to pass in from the component though, not the context or prop list.

Reply:
Done w/ latest commit